xla/service/gpu/transforms/post_layout_custom_call_rewriter.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright 2025 The OpenXLA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef XLA_SERVICE_GPU_TRANSFORMS_POST_LAYOUT_CUSTOM_CALL_REWRITER_H_
+#define XLA_SERVICE_GPU_TRANSFORMS_POST_LAYOUT_CUSTOM_CALL_REWRITER_H_
+
+#include "absl/container/flat_hash_set.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "xla/hlo/ir/hlo_module.h"
+#include "xla/hlo/pass/hlo_pass_interface.h"
+
+namespace xla {
+namespace gpu {
+
+// Rewrites custom calls after running layout assignment and normalization.
```

Comment:
Can you provide some details on the transformation that the rewriter performs?
It set up the memory space in the operand and use the operand to replace the custom call?